### ISSUE TITLE
Update "Secure" and "Insecure" language in example workflows

### ISSUE THREAD

**Issue by ericharmeling** [cockroachdb/docs#7816]
The workflow in many of our examples is split into "Secure" and "Insecure". We should update this language to emphasize that the "`--insecure`" `cockroach` mode is only meant for local development purposes, and should NEVER be used in production environments.

Suggested replacements:

- "Production" and "Development"
- "Secure" and "Testing"

@vy-ton @Amruta-Ranade @kenliu @thtruo 

Feel free to add to this list, or change it!

@taroface 
This will affect deployment docs as well.

**Comment by vy-ton** [comment]
+1 to `Production` and `Development`

**Comment by jseldess** [comment]
Thanks for tacking this, all. I think this is heading the right direction. I like using the production/testing terminology for deployment docs. I don't like that for our getting started docs and learning-centric tutorials because they are definitely not focused on production. 

Instead, I'd suggest we consider a different angle: Cloud vs. Local. Although we're not currently doing this, I think the right move is to default to having users leverage a free CC cluster. This supports our Product objective of "Make CockroachCloud the developer's database of choice" very well. The secondary option could be "Local", and there we could just provide guidance for a secure local cluster. For app dev tutorials, I think the best approach may be to just use `cockroach demo`, but I do wonder if persisting data is important to devs getting started. 

Two options is enough, but the third option we could provide very soon would be "In-Browser", leveraging the in-browser interactive tutorial approach I'm working on. 

cc @nstewart

**Comment by vy-ton** [comment]
For tutorials and developer recipes, my angle is to focus on the application developer mindset when approaching certain developer tasks. Having the `secure vs insecure` distinction at the top level is misleading because oftentimes that distinction only applies to specific steps, such as connecting to the database. 

Most third-party tool tutorials have the following introduction
```
Before you begin
1. Install CockroachDB.
2. Start up a secure or insecure local cluster.
3. Choose the instructions that correspond to whether your cluster is secure or insecure:
```
For starting the cluster (2), what options should we highlight? `Cloud vs local` would make sense here. But for the remaining content on the page, there is no or little difference between `cloud vs local` so "choose the instructions" (3) no longer applies.

For docs, I hope all code samples could one-day run in-browser.

**Comment by Amruta-Ranade** [comment]
+1 for `Cloud v. local`. 
@ericharmeling and I chatted about the app tutorials and making the instructions platform-agnostic:

1. Install CockroachDB.
2. Start a local cluster or spin up a CockroachCloud cluster. //the assumption here being local=secure
3. Connect to your cluster. //for CC, generate a connection string and for local, get the connection URL.

**Comment by jseldess** [comment]
@vy-ton, @Amruta-Ranade, I agree that the secure/insecure distinction is overblown right now and applies to only specific steps. Those steps do have to be exactly right, however, in order for these tutorials to be seamless. 

In any case, I don't think we should feel limited by the current structure of these tutorials. Right now, looking a the language-specific tutorials, most of these have a top-level toggle for the driver/orm, then a toggle for secure/insecure under "Before you begin". The driver/ORM toggles are actually links to distinct pages. We could imagine other approaches, for example:

- We could keep the top-level driver/ORM toggle, and make the primary "Before you begin" toggle "Cloud / In-Browser / Local". The rest of the page would redraw based on that decision. I will test this out as part of my Katacoda experiments next.
- I'm sure we could come up with other flows for these pages.

**Comment by ericharmeling** [comment]
`cockroach demo` is very useful for users looking to try out CRDB SQL syntax, but I'm finding it to complicate rather than simplify the local testing connection experience for developers.

It's been increasingly difficult for me to connect to `cockroach demo` clusters through "fully-supported" client drivers, primarily due to the security particularities of `cockroach demo`. Of the past 3 drivers (pgx, psycopg2, node-postgres) that I have tested, I get pretty much the same SSL certificate verification error (e.g., for psycopg2: `psycopg2.OperationalError: SSL error: certificate verify failed`). We could of course write our sample apps and tutorials to specify the directory with the temporary certificate, but that would add an additional step and an additional line of explanation ("`cockroach demo` generates a certificate at X directory. To specify the cert directory for your driver, do Y.", etc.). In the interim, I've opted for having users copy the full SQL unix domain socket URI (listed under `(sql/unix)` in 21.1+), but this only works for some of the drivers (notably, not for psycopg2).

I'm going to close this issue, with the easiest and most probable solution being to revert back to using an insecure single-node testing cluster, started with `cockroach start-single-node --insecure`. This is what the `cockroach demo --insecure` command (now deprecated) help recommends for testing. I can then add a warning about it being insecure, for testing purposes only, etc.
